src/PermissionsContext.tsx

```diff
@@ -18,40 +18,38 @@ import { isPlatformiOS } from "./utils"
 
 type ENAuthorizationStatus = `UNAUTHORIZED` | `AUTHORIZED`
 type ENEnablementStatus = `DISABLED` | `ENABLED`
-export type ENPermissionStatus = [ENAuthorizationStatus, ENEnablementStatus]
-const initialENPermissionStatus: ENPermissionStatus = [
+export type RawENPermissionStatus = [ENAuthorizationStatus, ENEnablementStatus]
+const initialENPermissionStatus: RawENPermissionStatus = [
   "UNAUTHORIZED",
   "DISABLED",
 ]
 
-export enum ENStatus {
-  UNAUTHORIZED_DISABLED,
-  AUTHORIZED_DISABLED,
-  AUTHORIZED_ENABLED,
+export enum ENPermissionStatus {
+  NOT_AUTHORIZED,
+  DISABLED,
+  ENABLED,
 }
 
-const toENStatus = (enPermissionStatus: ENPermissionStatus): ENStatus => {
+const toENPermissionStatusEnum = (
+  enPermissionStatus: RawENPermissionStatus,
+): ENPermissionStatus => {
   const isAuthorized = enPermissionStatus[0] === "AUTHORIZED"
   const isEnabled = enPermissionStatus[1] === "ENABLED"
 
-  if (!isAuthorized && !isEnabled) {
+  if (!isAuthorized) {
     return isPlatformiOS()
-      ? ENStatus.UNAUTHORIZED_DISABLED
-      : ENStatus.AUTHORIZED_DISABLED
+      ? ENPermissionStatus.NOT_AUTHORIZED
```

Comment:
@juancruzgs @johnschoeman Looks like my last comment was deleted due to a rebase. Is there a way to always return an authorized state on Android to make this check unnecessary?

Reply:
Merging this because it works but would be good to sort this out sometime.